src/Microsoft.AspNetCore.Cors/Infrastructure/CorsMiddleware.cs

```diff
@@ -18,18 +19,21 @@ public class CorsMiddleware
         private readonly ICorsPolicyProvider _corsPolicyProvider;
         private readonly CorsPolicy _policy;
         private readonly string _corsPolicyName;
+        private readonly ILogger _logger;
 
         /// <summary>
         /// Instantiates a new <see cref="CorsMiddleware"/>.
         /// </summary>
         /// <param name="next">The next middleware in the pipeline.</param>
         /// <param name="corsService">An instance of <see cref="ICorsService"/>.</param>
         /// <param name="policyProvider">A policy provider which can get an <see cref="CorsPolicy"/>.</param>
+        /// <param name="loggerFactory">An instance of <see cref="ILoggerFactory"/>.</param>
         public CorsMiddleware(
             RequestDelegate next,
             ICorsService corsService,
-            ICorsPolicyProvider policyProvider)
-            : this(next, corsService, policyProvider, policyName: null)
+            ICorsPolicyProvider policyProvider,
+            ILoggerFactory loggerFactory)
```

Comment:
AppVeyor is warning about API signature breaks here.
https://ci.appveyor.com/project/aspnetci/cors/build/1.0.476

For this 2.2 milestones we cannot change any existing APIs, only add new overloads.

Reply:
See what I did over here:
https://github.com/aspnet/BasicMiddleware/pull/340/files